numba/np/arraymath.py

```diff
@@ -809,110 +809,56 @@ def flat_all(a):
-    is_b_scalar = isinstance(b, types.Number)
-
-    if is_a_scalar and is_b_scalar:
-        def np_allclose_impl_scalar_scalar(a, b, rtol=1e-05, atol=1e-08,
-                                           equal_nan=False):
-            return _allclose_scalars(a, b, rtol=rtol, atol=atol,
-                                     equal_nan=equal_nan)
-        return np_allclose_impl_scalar_scalar
-    elif is_a_scalar and not is_b_scalar:
-        def np_allclose_impl_scalar_array(a, b, rtol=1e-05, atol=1e-08,
-                                          equal_nan=False):
-            b = np.asarray(b)
-            for bv in np.nditer(b):
-                if not _allclose_scalars(a, bv.item(), rtol=rtol, atol=atol,
-                                         equal_nan=equal_nan):
-                    return False
-            return True
-        return np_allclose_impl_scalar_array
-    elif not is_a_scalar and is_b_scalar:
-        def np_allclose_impl_array_scalar(a, b, rtol=1e-05, atol=1e-08,
-                                          equal_nan=False):
-            a = np.asarray(a)
-            for av in np.nditer(a):
-                if not _allclose_scalars(av.item(), b, rtol=rtol, atol=atol,
-                                         equal_nan=equal_nan):
-                    return False
-            return True
-        return np_allclose_impl_array_scalar
-    elif not is_a_scalar and not is_b_scalar:
-        def np_allclose_impl_array_array(a, b, rtol=1e-05, atol=1e-08,
-                                         equal_nan=False):
-            a = np.asarray(a)
-            b = np.asarray(b)
-            a_a, b_b = np.broadcast_arrays(a, b)
+    if a_isnan and b_isnan:
+        return equal_nan
 
-            for av, bv in np.nditer((a_a, b_b)):
-                if not _allclose_scalars(av.item(), bv.item(), rtol=rtol,
-                                         atol=atol, equal_nan=equal_nan):
-                    return False
+    if np.isinf(a) or np.isinf(b):
+        return a == b
 
-            return True
+    return np.abs(a - b) <= atol + rtol * np.abs(b * 1.0)
 
-        return np_allclose_impl_array_array
 
+@overload(np.allclose)
+def np_allclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
+    def impl(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
+        for av, bv in np.nditer((a, b)):
+            if not _isclose_scalars(av.item(), bv.item(), rtol=rtol,
+                                    atol=atol, equal_nan=equal_nan):
+                return False
+        return True
+    return impl
 
-@overload(np.any)
-@overload_method(types.Array, "any")
-def np_any(a):
-    def flat_any(a):
-        for v in np.nditer(a):
-            if v.item():
-                return True
-        return False
 
-    return flat_any
+@overload(np.isclose)
+def np_isclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
+    def impl(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
+        ab, ba = np.broadcast_arrays(a, b)
```

Comment:
rather than calling `np.broadcast_arrays`, compute the final shape with `np.broadcast_shapes` and call `np.broadcast_arrays` in each array individually. This might solve the current CI issue.

Reply:
I would also note that `np.nditer` doesn't work quite the same as the true numpy version; my use of it here seems to fail. I have to manually broadcast the arrays.

Reply:
> Hold on, why can't broadcast_arrays just return a Tuple(ArrayType1, ArrayType2)?

We follow NumPy spec where `np.broadcast_arrays` returns a list.